### PR TITLE
Add OIDC check

### PR DESCRIPTION
**Pull request by tsmithv11**

### CODE REVIEW

File: code-security/policy-reference/aws-policies/aws-iam-policies/aws-iam-policies.adoc [hunk 1]
```diff
@@ -155,5 +155,10 @@
 |MEDIUM
 
 
+|xref:iam-358.adoc[Ensure GitHub Actions OIDC trust policies only allows actions from a specific known organization]
```
Comment (gruebel) [inline code-security/policy-reference/aws-policies/aws-iam-policies/aws-iam-policies.adoc]:
```suggestion
|xref:iam-358.adoc[Ensure GitHub Actions OIDC trust policies only allow actions from a specific known organization]
```

File: code-security/policy-reference/aws-policies/aws-iam-policies/iam-358.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
+== Ensure GitHub Actions OIDC trust policies only allows actions from a specific known organization
```
Comment (gruebel) [inline code-security/policy-reference/aws-policies/aws-iam-policies/iam-358.adoc]:
```suggestion
== Ensure GitHub Actions OIDC trust policies only allow actions from a specific known organization
```

File: code-security/policy-reference/aws-policies/aws-iam-policies/iam-358.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+=== Fix - Buildtime
+
+
+*Terraform* 
+
+
+* *Resource: aws_iam_policy_document*
+* *Argument: principals, condition* 
+
+If you are using federated acces for GitHub Actions, add a `condition` with a specific repository.
```
Comment (gruebel) [inline code-security/policy-reference/aws-policies/aws-iam-policies/iam-358.adoc]:
```suggestion
If you are using federated access for GitHub Actions, add a `condition` with a specific organization.
```

File: code-security/policy-reference/aws-policies/aws-iam-policies/iam-358.adoc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      identifiers = ["arn:aws:iam::123456123456:oidc-provider/token.actions.githubusercontent.com"]
+      type        = "Federated"
+    }
+
++   # Add this block with your repository in the values
++   condition {
++     test     = "StringEquals"
++     values   = ["repo:myOrg/myRepo:ref:refs/heads/MyBranch"]
++     variable = "token.actions.githubusercontent.com:sub"
++   }
```
Comment (gruebel) [inline code-security/policy-reference/aws-policies/aws-iam-policies/iam-358.adoc]:
```suggestion
    condition {
      test     = "StringEquals"
      values   = ["sts.amazonaws.com"]
      variable = "token.actions.githubusercontent.com:aud"
    }

+   # Add this block with your organization name in the values
+   condition {
+     test     = "StringEquals"
+     values   = ["repo:myOrg/*"]
+     variable = "token.actions.githubusercontent.com:sub"
+   }
```
the next check will deal with the more restrictive approach of setting it to a specific branch

**Comment by Pubs-MV** [comment]
@anaghapamidi can I merge this?

**Review by gruebel — APPROVED** [review]

**Review by Pubs-MV — APPROVED** [review]

**Pull request merged**